compiler/rustc_passes/src/dead.rs

```diff
@@ -511,6 +512,11 @@ fn check_item<'tcx>(
         worklist.push(id.owner_id.def_id);
     }
 
+    let parent = tcx.local_parent(id.owner_id.def_id);
+    if matches!(tcx.def_kind(parent), DefKind::Closure | DefKind::Generator) {
```

Comment:
I think this is still the wrong place to put this check -- why not just add it to the match below? Look at the `DefKind::GlobalAsm` arm for example.

Reply:
Correction -- actually, closures aren't items. Maybe put this check in `visit_expr` instead? It just feels weird to be checking the parent instead of the closure itself when considering closures to be unconditionally live.

Reply:
Makes sense - moved into the expr check